File: FrameworkManager/sample/packages/ProjectPackage/iOSSample/Project/Classes/Model/TimelineModelBase.m
//
//  TimelineModelBase.m
//
//  Copyright (c) 2014年 saimushi. All rights reserved.
//

#import "TimelineModelBase.h"
#import "ProfileModel.h"


@implementation TimelineModelBase
{
    BOOL text_replaced;
    BOOL profile_id_replaced;
    BOOL owner_id_replaced;
    BOOL created_replaced;
    BOOL modified_replaced;
    BOOL available_replaced;

}

@synthesize text;
@synthesize profile_id;
@synthesize profile;
@synthesize owner_id;
@synthesize created;
@synthesize modified;
@synthesize available;


-(void)setText:(NSString *)argText
{
    text = argText;
    text_replaced = YES;
    replaced = YES;
}

-(void)setProfile_id:(NSString *)argProfile_id
{
    profile_id = argProfile_id;
    profile_id_replaced = YES;
    replaced = YES;
}

-(void)setOwner_id:(NSString *)argOwner_id
{
    owner_id = argOwner_id;
    owner_id_replaced = YES;
    replaced = YES;
}

-(void)setCreated:(NSString *)argCreated
{
    created = argCreated;
    created_replaced = YES;
    replaced = YES;
}

-(void)setModified:(NSString *)argModified
{
    modified = argModified;
    modified_replaced = YES;
    replaced = YES;
}

-(void)setAvailable:(NSString *)argAvailable
{
    available = argAvailable;
    available_replaced = YES;
    replaced = YES;
}



/* オーバーライド */
- (id)init:(NSString *)argProtocol :(NSString *)argDomain :(NSString *)argURLBase :(NSString *)argTokenKeyName;
{
    self = [super init:argProtocol :argDomain :argURLBase :argTokenKeyName];
    if(nil != self){
        modelName = @"timeline";
        text_replaced = NO;
        profile_id_replaced = NO;
        owner_id_replaced = NO;
        created_replaced = NO;
        modified_replaced = NO;
        available_replaced = NO;

    }
    return self;
}

/* オーバーライド */
- (BOOL)save;
{
    if(YES == replaced){
        NSMutableDictionary *saveParams = [[NSMutableDictionary alloc] init];
        if(YES == text_replaced){
            [saveParams setValue:self.text forKey:@"text"];
        }
        if(YES == profile_id_replaced){
            [saveParams setValue:self.profile_id forKey:@"profile_id"];
        }
        if(YES == owner_id_replaced){
            [saveParams setValue:self.owner_id forKey:@"owner_id"];
        }
        if(YES == created_replaced){
            [saveParams setValue:self.created forKey:@"created"];
        }
        if(YES == modified_replaced){
            [saveParams setValue:self.modified forKey:@"modified"];
        }
        if(YES == available_replaced){
            [saveParams setValue:self.available forKey:@"available"];
        }

        return [super _save:saveParams];
    }
    // 何もしないで終了
    return YES;
}

- (NSMutableDictionary *)convertModelData;
{
    NSMutableDictionary *newDic = [[NSMutableDictionary alloc] init];
    [newDic setObject:self.ID forKey:@"id"];
    [newDic setObject:self.text forKey:@"text"];
    [newDic setObject:self.profile_id forKey:@"profile_id"];
    /* ProfileモデルのDEEP-REST */
    NSMutableArray *profileList = [[NSMutableArray alloc] init];
    if(0 < self.profile.total){
        do {
            [profileList addObject:[self.profile convertModelData]];
        } while (YES == [self.profile next]);
    }
    [newDic setObject:profileList forKey:@"profile"];
    [newDic setObject:self.owner_id forKey:@"owner_id"];
    [newDic setObject:self.created forKey:@"created"];
    [newDic setObject:self.modified forKey:@"modified"];
    [newDic setObject:self.available forKey:@"available"];

    return newDic;
}

- (void)_setModelData:(NSMutableDictionary *)argDataDic;
{
    self.ID = [argDataDic objectForKey:@"id"];
    self.text = [argDataDic objectForKey:@"text"];
    self.profile_id = [argDataDic objectForKey:@"profile_id"];
    /* ProfileモデルのDEEP-REST */
    NSMutableArray *profileDic = [argDataDic objectForKey:@"profile"];
    if(nil != profileDic){
        self.profile = [[ProfileModel alloc] init:protocol :domain :urlbase :tokenKeyName :cryptKey :cryptIV :timeout];
        [self.profile setModelData:[argDataDic objectForKey:@"profile"]];
    }
    self.owner_id = [argDataDic objectForKey:@"owner_id"];
    self.created = [argDataDic objectForKey:@"created"];
    self.modified = [argDataDic objectForKey:@"modified"];
    self.available = [argDataDic objectForKey:@"available"];

    [self resetReplaceFlagment];
}

- (void)resetReplaceFlagment;
{
    text_replaced = NO;
    profile_id_replaced = NO;
    owner_id_replaced = NO;
    created_replaced = NO;
    modified_replaced = NO;
    available_replaced = NO;

    replaced = NO;
    return;
}

@end

File: FrameworkManager/sample/packages/AndroidProject/Project/app/src/main/java/jp/co/project/model/base/TimelineModelBase.java
package jp.co.project.model.base;

import android.content.Context;
import android.os.Handler;
import android.provider.ContactsContract;

import com.unicorn.manager.DataManager;
import com.unicorn.model.ModelBase;

import java.util.HashMap;

/**
 * プロフィール情報を扱う
 * 
 * @author c1718
 * 
 */
public class TimelineModelBase extends ModelBase {

	public static String TAG = "TimelineModel";
	/** ネーム */
	private String text;
	private boolean text_replaced;
	/** 作成日時 */
	private String created;
	private boolean created_replaced;
	/** 更新日時 */
	private String modified;
	private boolean modified_replaced;
	/** 有効・無効 */
	private String available;
	private boolean available_replaced;

	/**
	 *
	 * @param argContext コンテキスト
	 * @param argProtocol プロトコル
	 * @param argDomain ドメイン
	 * @param argURLBase ドメイン以下のディレクトリ名
	 * @param argTokenKeyName Cookieに保存するトークンのキー名
	 * @param argTimeout タイムアウトの時間
	 */
	public TimelineModelBase(Context argContext, String argProtocol, String argDomain,
							 String argURLBase, String argTokenKeyName, int argTimeout) {
		super(argContext, argProtocol, argDomain, argURLBase, argTokenKeyName, argTimeout);
		modelName = "timeline";
	}

	/**
	 *
	 * @param argContext コンテキスト
	 * @param argProtocol プロトコル
	 * @param argDomain ドメイン
	 * @param argURLBase ドメイン以下のディレクトリ名
	 * @param argTokenKeyName Cookieに保存するトークンのキー名
	 * @param argCryptKey 暗号化キー
	 * @param argCryptIV 初期化ベクトル
	 * @param argTimeout タイムアウトの時間
	 */
	public TimelineModelBase(Context argContext, String argProtocol, String argDomain,
							 String argURLBase, String argTokenKeyName, String argCryptKey, String argCryptIV,
							 int argTimeout) {
		super(argContext, argProtocol, argDomain, argURLBase, argTokenKeyName, argCryptKey,
				argCryptIV, argTimeout);
		modelName = "timeline";
	}

	/**
	 *
	 * @param argContext コンテキスト
	 * @param argProtocol プロトコル
	 * @param argDomain ドメイン
	 * @param argURLBase ドメイン以下のディレクトリ名
	 * @param argTokenKeyName Cookieに保存するトークンのキー名
	 * @param argCryptKey 暗号化キー
	 * @param argCryptIV 初期化ベクトル
	 */
	public TimelineModelBase(Context argContext, String argProtocol, String argDomain,
							 String argURLBase, String argTokenKeyName, String argCryptKey, String argCryptIV) {
		super(argContext, argProtocol, argDomain, argURLBase, argTokenKeyName, argCryptKey,
				argCryptIV);
		modelName = "timeline";
	}

	/**
	 *
	 * @param argContext コンテキスト
	 * @param argProtocol プロトコル
	 * @param argDomain ドメイン
	 * @param argURLBase ドメイン以下のディレクトリ名
	 * @param argTokenKeyName Cookieに保存するトークンのキー名
	 */
	public TimelineModelBase(Context argContext, String argProtocol, String argDomain,
							 String argURLBase, String argTokenKeyName) {
		super(argContext, argProtocol, argDomain, argURLBase, argTokenKeyName);
		modelName = "timeline";
	}

	/**
	 *
	 * @param argContext コンテキスト
	 */
	public TimelineModelBase(Context argContext) {
		super(argContext);
		modelName = "timeline";
	}

	@Override
	public void resetReplaceFlagment() {
		// replacedは親クラスでやった方がいいかも
		this.replaced = false;

		this.text_replaced = false;
		this.created_replaced = false;
		this.modified_replaced = false;
		this.available_replaced = false;
	}

	@Override
	public boolean save() {
		if (!this.replaced) {
			// 更新したことにする
			return true;
		}
		HashMap<String, Object> argsaveParams = new HashMap<String, Object>();
		argsaveParams.put("profile_id",DataManager.getInstance().getUserModel().ID);
		argsaveParams.put("owner_id",DataManager.getInstance().getUserModel().ID);

		if (this.text_replaced) {
			argsaveParams.put("text", this.text);
		}
		if (this.created_replaced) {
			argsaveParams.put("created", this.created);
		}
		if (this.modified_replaced) {
			argsaveParams.put("modified", this.modified);
		}
		if (this.available_replaced) {
			argsaveParams.put("available", this.available);
		}
		return super.save(argsaveParams);
	}

	public boolean save(Handler completeHandler) {
		completionHandler = completeHandler;
		return save();
	}

	@Override
	public void _setModelData(HashMap<String, Object> map) {

		if (!map.containsKey("id")) {
			// プロフィール情報が存在しないと判断
			return;
		}
		this.ID = String.valueOf(map.get("id"));
		this.text = String.valueOf(map.get("text"));
		this.created = String.valueOf(map.get("created"));
		this.modified = String.valueOf(map.get("modified"));
		this.available = String.valueOf(map.get("available"));
	}

	@Override
	public HashMap<String, Object> convertModelData() {
		HashMap<String, Object> newMap = new HashMap<String, Object>();
		newMap.put("id", ID);
		newMap.put("text", this.text);
		newMap.put("created", this.created);
		newMap.put("modified", this.modified);
		newMap.put("available", this.available);
		return newMap;
	}

	/**
	 * @return ネーム
	 */
	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
		this.text_replaced = true;
		this.replaced = true;
	}

	/**
	 * @return 更新日時
     */
	public String getModified() {
		return modified;
	}

}
